aplib.net-demo/Assets/Scripts/Inventory/Inventory.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Inventory : MonoBehaviour
+{
+    private Queue<Item> _itemList;
+    public float inventorySize;
+    /// <summary>
+    /// The RawImage is the object on which the icon texture is projected
+    /// </summary>
+    public RawImage icon;
+    /// <summary>
+    /// the texture of the icon object
+    /// </summary>
+    public Texture iconTexture;
+    public GameObject inventoryObject;
+
+    //here you would add all the possible items you can add, you add them to the inventory by calling PickUpItem method with the right item
+
+    /// <summary>
+    /// Creates the inventory queue and sets default size, resets the items, and fetches the rawimage component to display the icons
+    /// </summary>
+    private void Start()
+    {
+        icon = GetComponent<RawImage>();
+        _itemList = new Queue<Item>();
+    }
+
+    /// <summary>
+    /// Converts queue to list to check if there are any items with matching names. If there are it checks if they are stackable and adds uses. If they are not it does nothing. If there are not matching names it adds the item to the inventory;
+    /// </summary>
+    /// <param name="item">The item that is fed into the inventory</param>
+    /// <param name="uses">the amount of uses that are added upon pickup</param>
+    public void PickUpItem(Item item, float uses = 1)
+    {
+        float queueSize = _itemList.Count();
+        bool alreadyInInventory = false;
+        for (int i = 0; i < queueSize; i++)
+        {
+            if (_itemList.ToList()[i].name == item.name)
+            {
+                if (item.stackable)
+                {
+                    _itemList.ToList()[i].uses += uses;
+                    alreadyInInventory = true;
+                    break;
+                }
+                else
+                {
+                    DisplayItem();
+                    return;
```

Comment:
The displayitem was redundant, since the first item in the inventory doesn't change, the display doesn't need to be updated. I've removed that

if you want to pick up an item that is not stackable, and you don't have it, it equips as normal. If you already have it, it indeed consumes it (like if you already have a knife and pick it up, it doesn't really change if it stays there or is removed since it's not consumable).

Reply:
I would consider this confusing behaviour. What do others think?

Reply:
Moved this discussion to APLIB-58 Add Spawnable Items